### PR TITLE
ICU-21480 integrate CLDR release-39-alpha4 to ICU trunk

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21480
- [x] Required: The PR title must be prefixed with a JIRA Issue number. For example: "ICU-1234 Fix xyz"
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. For example: "ICU-1234 Fix xyz"
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

This integrates the CLDR data as of Feb 24 at 9 PM Pacific time into ICU. The CLDR data is not tagged yet; there is one test data file that comes from CLDR that needed to be updated for ICU tests to pass, and I am waiting for that to be merged into CLDR per https://github.com/unicode-org/cldr/pull/1075. Once that happens the CLDR data will be tagged as "release-39-alpha4".

This has more ICU changes than the past couple of integrations:
- The mixed metric changes in [CLDR-14515](https://unicode-org.atlassian.net/browse/CLDR-14515), picked up in this integration, required a number of test updates.
- This also correctly adjusts the Norwegian files in icu4c/source/data/xml/rbnf/, changing nb.xml to no.xml, and removing nn.xml. This was something that should have been done as part of [PR-1581](https://github.com/unicode-org/icu/pull/1581), but wasn't, leading to the "mangled data/rbnf/nb.txt" issue in [ICU-21496](https://unicode-org.atlassian.net/browse/ICU-21496); thanks to David Beaumont for pointing that out. That issue is now fixed with this PR.

### CODE REVIEW

File: icu4c/source/data/cldr-icu-readme.txt [hunk 1]
```diff
@@ -201,14 +201,6 @@ ant copy-cldr-testdata
 # and add the following line to the beginning of these two files
 # # File copied from cldr common/testData/localeIdentifiers/localeCanonicalization.txt
 
-# 4e. Temporary fix until ICU-21496 is addressed: Need to remove bogus content.
-cd $ICU4C_DIR/source/data
-open rbnf/nb.txt
```
Comment (hagbard) [inline icu4c/source/data/cldr-icu-readme.txt:206]: Just remove the file. Don't use the "specials" to try and define parent relations.

And why not move the file?

File: icu4c/source/data/misc/units.txt [hunk 2]
```diff
@@ -1140,21 +1135,6 @@ units:table(nofallback){
...
                     }
                 }
-                CN{
-                    {
-                        unit{"meter-and-centimeter"}
-                    }
-                }
-                DE{
-                    {
-                        unit{"meter-and-centimeter"}
```
Comment (hugovdm) [inline icu4c/source/data/misc/units.txt:1150]: Thanks for dealing with the unit test!

(I want synthetic testing data in ICU unit tests!)

**Review by hugovdm — APPROVED** [review]
Units changes look good, thanks! (Norwegian-related changes are still beyond my knowledge level.)

File: icu4c/source/data/cldr-icu-readme.txt (hunk 1 above)
Comment (pedberg-icu) [reply icu4c/source/data/cldr-icu-readme.txt:206]: Huh? Not sure what you are talking about here. We do need to have data/rbnf/nb.txt, but now it is generated correctly.

If you are talking about the data/xml/rbnf files for Norwegian, they are *not* being used to define parent relationships, that is being done in the CLDR parent data. They *are* being used to define ICU-specific RBNF additions; but now that additional data is only needed for "no", and inherited by "nb" and "nn".

**Pull request merged**